File: samples/random.c
#include <stdio.h>
#include <stdlib.h>
#define ROCK 0
#define PAPER 1
#define SCISSORS 2

#define PROGRAM_NAME "Sample Random C"

int* my_history;
int* opp_history;
int  turns_passed = 0;
int  game_length;

int get_next_move(){
	// do your fancy maths here
	return rand()%3;
}

int main(int argc, char** argv){
	int tmp;
	printf("%s\n", PROGRAM_NAME);
	fflush(stdout);
	scanf("%d", &tmp);
	srand(tmp);
	scanf("%d", &game_length);
	my_history = malloc(sizeof(int)*game_length);
	opp_history = malloc(sizeof(int)*game_length);
	while(1){
		tmp = get_next_move();
		printf("%d\n", tmp);
		my_history[turns_passed] = tmp;
		fflush(stdout);
		scanf("%d", opp_history+turns_passed);
		turns_passed ++;
	}
	return 0;
}

File: samples/random.hs
module Main where
import System.Random
import System.IO

type State = ()


programName = "Sample Random Haskell"

rock     = 0
paper    = 1
scissors = 2

initialState = ()

getNextMove :: RandomGen g => State -> [Int] -> [Int] -> g -> Int -> (Int, State, g)
getNextMove state myHistory oppHistory random gameLength = 
	let 
		-- do your fancy maths here
		newState = state
		(value, newRandom) = randomR (0,2) random
	in
		(value, newState, newRandom)

runGame state myHistory oppHistory random gameLength = do
	let (value, newState, newRandom) = getNextMove state myHistory oppHistory random gameLength
	putStrLn $ show value
	hFlush stdout
	h <- readLn
	runGame newState (value:myHistory) (h:oppHistory) newRandom gameLength

main = do
	putStrLn programName
	hFlush stdout
	seed <- readLn
	gameLength <- readLn
	runGame initialState [] [] (mkStdGen seed) gameLength
